packages/core/src/codecatalyst/devEnv.ts

```diff
@@ -11,6 +11,122 @@ import { Timeout } from '../shared/utilities/timeoutUtils'
 import { showMessageWithCancel } from '../shared/utilities/messages'
 import { isCloud9 } from '../shared/extensionUtilities'
 import { getLogger } from '../shared/logger'
+import { CodeCatalystAuthenticationProvider } from './auth'
+import { getThisDevEnv } from './model'
+import { isInDevEnv } from '../shared/vscode/env'
+import { shared } from '../shared/utilities/functionUtils'
+
+/** This class exists due to complexities with starting the Dev Env Activity Hearbeat mechanism */
+export class DevEnvActivityHeartbeatStarter {
```

Comment:
long names really can be counterproductive. shorter names help others by mentioning the most important aspects of a thing. repeating "DevEnv" in something that lives in devEnv.ts is probably avoidable. and "Starter" is over-explanation, because there is no other "ActivityHeartbear" class, so there is no ambiguity without "Starter".

```suggestion
export class ActivityHeartbeat {
```

Reply:
This makes sense, but this class only handles starting of the `DevEnvActivity` class.
I made this class to contain all the "starting" code in a single place. I've renamed to `DevEnvActivityStarter`, and update the docstring to better reflect this.